### ISSUE TITLE
Make generate-project-files idempotent

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1291]
**Is your feature request related to a problem? Please describe.**

Task creation may fail due to various reasons not related to incorrect input (server overload, db connection limit, etc).

*Describe the solution you'd like**

The project files generation should be idempotent. 

This means that there are checks to see which tasks have an odk token, form in odk central and uploaded geojson media. 

It would increase the time of the endpoint due to additional calls, but worth it to allow multiple sequential executions, i.e. resume.

**Describe alternatives you've considered**

An additional endpoint to resume, but it would be nice to have the same one.

**Comment by spwoodcock** [comment]
To facilitate this I think we should update the project creation flow.

**Current**: org admin creates entire project.
**Proposal**:
- Org admin create a project stub with minimal details, assign a project admin
  - Requires a new endpoint `/project/stub/` (or perhaps reuse `/project` POST endpoint.
- Project admin fills out details and submits. This calls the project update endpoint, which is by default idempotent.
  - The project update will fill out the remaining fields.
  - For this to work, we probably need a flag in the db to determine when a project is full filled out, or only a stub.

Advantages of this approach:
- Idemponent: project creation can be repeated without issues.
- Reduce code duplication: the editing workflow can be the same as the project creation.
- Permissions: better to pass main project creation responsibility to project admin.
- Ordering: project is created prior to data extract, meaning we can include **project** id in the data extract file name.
  - This allows for data extract tracking by HOT in production.

**Comment by spwoodcock** [comment]
Note: we have ProjectStatus enum with `DRAFT=2`, so should probably create the stub with DRAFT status.

Once complete, change to `PUBLISHED`, enum id = 1

**Comment by Anuj-Gupta4** [comment]
The post project API can already be called with only a few parameters:

- Name
- Description
- Other instructions
- Project AOI
- ODK credentials

We could let organization admin input these as a draft project. It's probably ok to ask for ODK credentials here because sometimes, we use organization's ODK credentials. Maybe we can have project admin later update them if they would prefer to use their own?

There are separate APIs to do everything else. We can update the access to be project manager only. Once we make sure tasks are successfully created and related files are generated, we can update the project status to be published.

**Comment by spwoodcock** [comment]
Yeah I think the project manager should be deciding on / adding ODK creds 👍 

I'm thinking it would be ideal if the org manager only needs to specify the project name as mandatory, but the project is created as `draft`.

- On the first page of the UI, they enter project name and (optional) project managers.
- Either:
  - Org admin wishes to continue, and continues to fill fields.
  - Org manager exits to allow project manager to do the rest.
    - The project mangers assigned to the project are notified, with a link to go straight to editing.
    - Any project managers or org managers can see the newly created project in the UI (with a 'Draft' tag).
    - If they click the draft project, they are redirected to continue project creation.

**Comment by manjitapandey** [comment]
I will summarize my understanding below:
1. Project name and description: This should be set by organization admin (mandatory). Description is necessary for project managers to understand what the project is about and can go on accordingly. Should it be editable by project managers afterwards??
2. Assign project manager: Organization admin should compulsorily assign project manager (if they want to add project as draft), but is optional if they will complete the whole project creation by themselves.
3. Organization: Choosing organization should be restricted to org admins 
4. Project AOI: This should also be made compulsory by org admin since project name can be editable and we need something that can't be changed afterwards. 

Other things are clear to add them as draft and project managers can go ahead with project editing filling up necessary fields and task splitting.

**Comment by spwoodcock** [comment]
Yep sounds good to me 👍

So we can make these fields mandatory for org manager in the first step:

- name
- description
- outline

Then optional:

- Assign project managers and save for later
- Continue adding all fields

**Comment by manjitapandey** [comment]
Figma Design of stub project creation workflow:

![Image](https://github.com/user-attachments/assets/e295867c-77c9-455a-8129-5484b8e42394)

![Image](https://github.com/user-attachments/assets/aa03ed49-0e8f-431c-9ea7-a8dae25b58e0)

There are no major changes on later steps. UI components are mostly the same. 
Please provide your feedback and we can start the development. 
cc: @spwoodcock , @NSUWAL123 , @ramyaragupathy

**Comment by spwoodcock** [comment]
This looks nice!

It would capture 2 flows:
- Draft project (to be picked up by assigned admin) via API call to stub project creation endpoint
- Full project creation by two API calls. First the stub / basic draft details, then the project update endpoint to add the remaining details and set as published

**Comment by manjitapandey** [comment]
> Do we have the functionality to auto notify assigned project managers either via email or a notification within the app?In the future may b this could be listed via a dedicated user dashboard. But for now having these notifications could work.

There is no such notification within the field-TM platform but we inform assigned project managers via email and notification on openstreetmap. 

![Image](https://github.com/user-attachments/assets/57847d2c-2bc8-417e-813d-046aa768d258)

For second and third suggestions, thank you, we will consider that.

**Comment by manjitapandey** [comment]
Hi @ramyaragupathy, here is the updated design of project creation considering your suggestion. Please let us know what you think.

cc: @NSUWAL123 

![Image](https://github.com/user-attachments/assets/9c1b5240-dd6c-4783-89e6-e01637869685)
